src/WorkflowEngine.Sdk/Support/WorkflowHelper.cs

```diff
@@ -130,4 +130,40 @@ public static Task WhenAllActivities(params Task[] tasks)
         return WhenAllActivities(tasks.ToList());
     }
 
+    /// <summary>
+    /// Performs a retry operation for given function.
+    /// </summary>
+    /// <param name="actionToExecute">Function to execute.</param>
+    /// <param name="maxRetries">MAximum number of retrys.</param>
+    /// <param name="delayBetweenRetriesInMilliseconds">Deley between retrys.</param>
+    /// <param name="reThrowException">If end result is an exception this flag determins if exception will be re-thrown.</param>
+    /// <returns></returns>
+    public static async Task RetryAsync(
+        Func<Task> actionToExecute,
+        int maxRetries,
+        int delayBetweenRetriesInMilliseconds,
+        bool reThrowException = false)
+    {
+        int retryCount = 0;
+        while (true)
+        {
+            try
+            {
+                await actionToExecute();
+                return;
+            }
+            catch (Exception)
+            {
+                if (retryCount >= maxRetries && reThrowException)

```

Comment:
If reThrowException is false, then we will have an eternal loop?

Reply:
I would prefer if it always rethrows if it doesn't succeed in maxRetries, so that it is shown in the calling method that we ignore the exceptions.